test/run_all_tests.sh

```diff
@@ -2,6 +2,12 @@
 
 all_tests_result=0
 
+if [ -z "$COVERALLS_REPO_TOKEN" ]; then
```

Comment:
This may be premature. Seems like it would interfere with someone who is iterating on a feature and only cared to run the unit-tests, but not submit a coverage report to coveralls.io.

Reply:
We should probably add a flag (e.g. `--report-coverage`) to the `run_all_tests.sh` script. This would allow a user to easily use the script during iterative development, but also allow the CI to publish stats to Coveralls.io.

This could also be handled by moving the submission of the coverage statistics to Coveralls.io into a subsequent GitHub action (build step).

Reply:
GMTA! I was thinking this should ideally be a separate step.  That way you only pay for what you need.  Also avoids treading on iterative development.  I'll factor out the coveralls publish to a separate step, with its own script for local development and a separate github action.